src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -66,4 +80,105 @@ public String version() {
+     *
+     * For a valid endpoint and a valid token, this returns a HTTP 400 Bad Request with the
+     * payload: {"text":"No data","code":5}
+     * For a valid endpoint and an invalid token, this returns a HTTP 403 Forbidden with the
+     * payload: {"text":"Invalid token","code":4}
+     * For an invalid hostname and other errors, the Java UnknownHostException and other similar
+     * Exceptions are thrown.
+     *
+     * @param connectorConfig The connector configuration
+     * @param configValues The configuration ConfigValues
+     */
+    private void validateAccess(SplunkSinkConnectorConfig connectorConfig, Map<String, ConfigValue> configValues) {
+        HecConfig config = connectorConfig.getHecConfig();
+
+        try (CloseableHttpClient httpClient = createHttpClient(config)) {
+            List<String> uris = config.getUris();
+            final String hecToken = config.getToken();
+
+            CountDownLatch latch = new CountDownLatch(config.getUris().size());
+            Map<String, String> validationFailedIndexers = new LinkedHashMap<>();
+
+            for (String uri : uris) {
+                log.info("Validating " + uri);
+                HttpPost request = new HttpPost(uri + "/services/collector");
+                request.setEntity(new StringEntity(""));
+
+                request.addHeader(HttpHeaders.AUTHORIZATION, String.format("Splunk %s", hecToken));
+
+                int status = -1;
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    status = response.getStatusLine().getStatusCode();
+                    if (status == 400) {
+                        log.info("Validation succeeded for indexer {}", uri);
+                    } else if (status == 403) {
+                        log.error("Invalid HEC token for indexer {}", uri);
+                        validationFailedIndexers.put(uri, response.getStatusLine().toString());
+                    } else {
+                        log.error("Validation failed for {}", uri);
+                        validationFailedIndexers.put(uri, response.getStatusLine().toString());
+                    }
+                } catch (Exception e) {
+                    log.error("Caught exception while validating", e);
+                    validationFailedIndexers.put(uri, e.getMessage());
+                } finally {
+                    latch.countDown();
+                }
+            }
+
+            latch.await(CONFIG_VALIDATION_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+            if (!validationFailedIndexers.isEmpty()) {
+                log.error("Validation failed: " + validationFailedIndexers);
+                configValues.get(SplunkSinkConnectorConfig.URI_CONF)
+                    .addErrorMessage("Validation Failed: " + validationFailedIndexers);
+                configValues.get(SplunkSinkConnectorConfig.TOKEN_CONF)
+                    .addErrorMessage("Validation Failed: " + validationFailedIndexers);
```

Comment:
optional: We can move these calls to a common useful function like `recordErrors(String error, String ...configNames, Config configValues)`. This would also make the code more readable IMO.

Reply:
Sure, done!